src/dataFlattener.ts

```diff
@@ -0,0 +1,66 @@
+// Needs typing (I originally wrote this in javascript)
+
+export const dataFlattener = (nestedData) => {
+  type flatData = {
+    code: string;
+    name: string;
+    type: string;
+    country: string;
+    subnational1: string | null;
+    subnational2: string | null;
+    fullHierarchyName: string;
+  };
+
+  type region = {
+    code: string;
+    name: string;
+    subnational1: string | undefined;
+    subnational2: string | undefined;
+  };
+
+  const flattenedData: flatData[] = [];
+
+  function recurse(
+    region: region,
+    country: string,
+    subdivision1 = null,
+    subdivision2 = null
+  ) {
+    const { code, name, subnational1, subnational2 } = region;
+
+    let type = "country";
+    if (subnational1 !== undefined && subnational2 === undefined) {
+      country = name;
```

Comment:
I would _strongly_ suggest that we avoid mutating data at all (i.e taking an existing variable and re-assigning its value) and as a general rule we should definitely avoid mutating the function arguments. 

It can be quite dangerous and surprising to a consumer to call your function and have a bit of data they thought they owned change into something else.

In this instance I don't think it causes problems but in general it can produce bugs that are extremely difficult to identify and resolve. 

We can definitely make a function here that doesn't need to mutate its arguments in order to work. We might not even need to make it recursive.